src/fastcs_odin/odin_adapter_controller.py

```diff
@@ -79,13 +76,13 @@ class StatusSummaryUpdater(Updater):
 
     path_filter: list[str | tuple[str] | re.Pattern]
     attribute_name: str
-    accumulator: Callable[[Iterable[T]], T]
-    update_period: float = 0.2
+    accumulator: Callable[[Iterable[Any]], float | int | bool | str]
+    update_period: float | None = 0.2
 
     async def update(self, controller: "OdinAdapterController", attr: AttrR):
         values = []
         for sub_controller in _filter_sub_controllers(controller, self.path_filter):
-            sub_attribute: AttrR = getattr(sub_controller, self.attribute_name)
+            sub_attribute: AttrR = sub_controller.attributes[self.attribute_name]  # type: ignore
```

Comment:
How about we implement `Controller.{read,write}_attributes` dict views that would expose only attributes with the matching access mode and raise an exception if trying to access an AttrR via `write_attributes`?

Reply:
Added issue: https://github.com/DiamondLightSource/FastCS/issues/104